File: BluetoothLowEnergy/app/src/main/java/com/bignerdranch/android/bluetoothtestbed/client/GattServerViewModel.java
package com.bignerdranch.android.bluetoothtestbed.client;

import android.bluetooth.BluetoothDevice;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;

public class GattServerViewModel extends BaseObservable {

    private BluetoothDevice mBluetoothDevice;

    public GattServerViewModel(BluetoothDevice bluetoothDevice) {
        mBluetoothDevice = bluetoothDevice;
    }

    @Bindable
    public String getServerName() {
        if (mBluetoothDevice == null) {
            return "";
        }
        return mBluetoothDevice.getAddress();
    }
}

File: BluetoothLowEnergy/app/src/main/java/com/bignerdranch/android/bluetoothtestbed/client/GattServerViewModel.kt
package com.bignerdranch.android.bluetoothtestbed.client

import android.bluetooth.BluetoothDevice
import androidx.databinding.BaseObservable
import androidx.databinding.Bindable

class GattServerViewModel(private val mBluetoothDevice: BluetoothDevice?) : BaseObservable() {
    @get:Bindable val serverName: String
        get() = mBluetoothDevice?.address ?: ""
}